report_viewer/apps/viewer/tests.py

```diff
@@ -169,3 +174,65 @@ def test_view_report_not_on_s3(client):
     assert response.status_code == 200
 
     assertContains(response, html_on_db)
+
+
+def test_report_metric_middleware_client_ip():
+    get_response = mock.MagicMock()
+    factory = RequestFactory()
+    request = factory.get("/")
+    rm = ReportMetrics(get_response)
+    res = rm.client_ip(request)
+    assert res == "127.0.0.1"
+
+
+def test_report_metric_middleware_string_to_hash():
+    os.environ["SECRET_KEY"] = "12345678"
+    get_response = mock.MagicMock()
+    factory = RequestFactory()
+    request = factory.get("/", HTTP_USER_AGENT="Mozilla/5.0")
+    rm = ReportMetrics(get_response)
+    res = rm.string_to_hash(request)
+    assert res == "Mozilla/5.0127.0.0.112345678"
```

Comment:
When I run this test locally the value of `res` is `Mozilla/5.0127.0.0.19DqDW2I6FvOA2cACDJr89DqDW2I6FvOA2cACDJr8`.

Reply:
fixed